### PR TITLE
Feat/packet forwarder

### PR DESCRIPTION
**Pull request by scartill**
Added support for Semtech `packet forwarder` protocol to be used with a generic network server (not only chirpstack), which receives data via UDP. To the date, only unconfirmed uplinks are supported. UDP reception could be quite easily implemented as well, if needed by anyone.

BTW, thank you for a great tool.

### CODE REVIEW

**Comment by iegomez** [comment]
Hi, @scartill, thanks for the PR! Let me take a look at it and I'll get back to you as soon as I can.

File: README.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ It has a simple but complete GUI built with [imgui-go](https://github.com/inkybl
 
 Please report any bug or request new features by filing and issue.
 
+This program was also extended to generate raw PACKET FORWARDER UDP-based protocol as alternative transport, to be used with general LoRaWAN netfork server (e.g. [lorawan-server](https://github.com/gotthardp/lorawan-server)). Use `forwarder` configuration section to enable.
```
Comment (iegomez) [inline README.md]: There's a typo in `netfork`.

File: device.go [hunk 2]
```diff
@@ -496,7 +499,14 @@ func run() {
 		}
 
 		//Now send an uplink
-		ulfc, err := cDevice.Uplink(mqttClient, config.MQTT.UplinkTopic, config.Device.MType, uint8(config.RawPayload.FPort), &urx, &utx, payload, config.GW.MAC, config.Band.Name, dataRate, fOpts, fCtrl)
+		var ulfc uint32 = 0
```
Comment (iegomez) [inline device.go]: There's no need to assign `nil` to `err`. Also, prefer to declare `ulfc` like this:
```
var ulfc uint32
```

File: forwarder.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	beginOpenFile()
+	beginSaveFile()
+	imgui.End()
+}
+
+func forwarderConnect() error {
+	port, err := strconv.Atoi(config.Forwarder.Port)
+
+	if err != nil {
+		log.Warn("Bad Network server UDP port")
```
Comment (iegomez) [inline forwarder.go]: I think the message could be improved and prefer to start with a small case, something like this (just an example): network server UDP port must be a number.

File: forwarder.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		return err
+	}
+
+	cNSClient.Server = config.Forwarder.Server
+	cNSClient.Port = port
+	cNSClient.Connected = true
+	log.Infoln("UDP Forwarder Started (MQTT disabled)")
```
Comment (iegomez) [inline forwarder.go]: Use lower case for started.

File: forwarder.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	// TODO subscribe to downlinks
+
+	return nil
+}
+
+func forwarderDisconnect() error {
+	cNSClient.Connected = false
+	log.Infoln("UDP Forwarder Stopped (MQTT back again")
```
Comment (iegomez) [inline forwarder.go]: Same as above and you're missing a closing parenthesis.

**Review by iegomez — CHANGES_REQUESTED** [review]

**Comment by iegomez** [comment]
Hey, @scartill, sorry for the delay. This looks good, just left some nitpick comments.
By the way, I'm not working with anything LoRa related since almost half a year now so I don't get to use or test this any more. Would you like to be added as a maintainer so the repo stays alive?

**Comment by scartill** [comment]
Hello, @iegomez, sorry for a pause from my side. I fixed all the issues. I'm not sure that I will actively continue the development, but anyways I'll be glad to help, so add me to maintainers, please.

**Pull request merged**